test/e2e_node/system/kernel_validator_helper.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package system
+
+import (
+	"os/exec"
+)
+
+// KernelValidatorHelper is an interface intended to help with os specific kernel validation
+type KernelValidatorHelper interface {
+	// GetKernelRelease gets the current kernel release version of the system
+	GetKernelRelease() ([]byte, error)
+}
+
+// LinuxKernelValidatorHelper is the 'linux' implementation of KernelValidatorHelper
+type LinuxKernelValidatorHelper struct{}
```

Comment:
When thinking about it; you _could_ move this into the platform-specific files and call it `KernelValidatorHelperImpl`, and this time abstract away the OS differences easily. That would also reduce the need for the map.
One tip: when creating a class that implements an interface, write this:
```go
var _ Iface = &Impl{}
// Which in this case would be:
var _ KernelValidatorHelper = &LinuxKernelValidatorHelper{}
```

Reply:
Makes sense :+1: